core/src/type_info.rs

```diff
@@ -0,0 +1,304 @@
+    sql_data_type: SqlDataType,
+}
+
+impl MongoTypesInfo {
+    pub fn new(sql_data_type: SqlDataType) -> MongoTypesInfo {
+        MongoTypesInfo {
+            current_type_index: 0,
+            sql_data_type,
+        }
+    }
+}
+
+impl MongoStatement for MongoTypesInfo {
+    // iterate through the list, searching for the next "valid" data type.
+    // a type is valid if its sql type matches the desired sql type, or if we are getting all types.
+    fn next(&mut self, _: Option<&MongoConnection>) -> Result<bool> {
+        loop {
+            self.current_type_index += 1;
+            if self.current_type_index > (DATA_TYPES.len() as isize)
+                || DATA_TYPES[(self.current_type_index - 1) as usize].sql_type == self.sql_data_type
+                || self.sql_data_type == SqlDataType(0)
+            {
+                break;
+            }
+        }
+        Ok(self.current_type_index <= (DATA_TYPES.len() as isize))
+    }
+
+    // Get the BSON value for the cell at the given colIndex on the current row.
+    fn get_value(&self, col_index: u16) -> Result<Option<Bson>> {
+        // 1 -> TYPE_NAME
+        // 2 -> DATA_TYPE
+        // 3 -> COLUMN_SIZE
+        // 4 -> LITERAL_PREFIX
+        // 5 -> LITERAL_SUFFIX
+        // 6 -> CREATE_PARAMS
+        // 7 -> NULLABLE
+        // 8 -> CASE_SENSITIVE
+        // 9 -> SEARCHABLE
+        // 10 -> UNSIGNED_ATTRIBUTE
+        // 11 -> FIXED_PREC_SCALE
+        // 12 -> AUTO_UNIQUE_VALUE
+        // 13 -> LOCAL_TYPE_NAME
+        // 14 -> MINIMUM_SCALE
+        // 15 -> MAXIMUM_SCALE
+        // 16 -> SQL_DATA_TYPE
+        // 17 -> SQL_DATETIME_SUB
+        // 18 -> NUM_PREC_RADIX
+        // 19 -> INTERVAL_PRECISION
+        match DATA_TYPES.get((self.current_type_index - 1) as usize) {
+            Some(type_info) => {
+                Ok(Some(match col_index {
+                    1 | 13 => Bson::String(type_info.type_name.to_string()),
+                    2 | 16 => Bson::Int32(type_info.sql_type.0 as i32),
+                    3 => match type_info.precision {
+                        Some(precision) => Bson::Int32(precision as i32),
+                        None => Bson::Null,
+                    },
+                    4 | 5 => match **type_info {
+                        BsonTypeInfo::DATE | BsonTypeInfo::STRING => Bson::String("'".to_string()),
+                        _ => Bson::Null,
+                    },
+                    6 => Bson::Null,
+                    7 => Bson::Boolean(**type_info == BsonTypeInfo::OBJECTID),
+                    8 => match **type_info {
+                        BsonTypeInfo::STRING => Bson::Int32(1),
+                        _ => Bson::Null,
+                    },
+                    9 => {
+                        match type_info.searchable {
+                            // SQL_SEARCHABLE and SQL_PRED_NONE, respectively
+                            true => Bson::Int32(3),
+                            false => Bson::Int32(0),
```

Comment:
```suggestion
                        match type_info.searchable {
                            // SQL_SEARCHABLE and SQL_PRED_NONE, respectively
                            true => Bson::Int32(something::SQL_SEARCHABLE),
                            false => Bson::Int32(something::SQL_PRED_NONE),
```
Can we possibly import those values from somewhere so they are clearly named within the code itself? The comment is probably good enough, but named variables would be great.

Reply:
I think this is a good idea, can move these to `definitions.rs`